### PR TITLE
Refactor buttons

### PR DESCRIPTION
**Pull request by farooqkz**
Buttons have been refactored into a single `Button` component. The component is then used everywhere there is `delta-button`. In the previous PR which is now closed, adz made some points:

>For buttons inside DialogFooter (former DeltaDialogFooter) I've introduced the FooterActionButton (https://github.com/deltachat/deltachat-desktop/blob/master/src/renderer/components/Dialog/FooterActionButton.tsx) component which looks very similar to this one here.
> Currently all buttons in dialogues are always bold and primary, so no need to be highly configurable in this context.
> When rebasing this PR to master (with my refactorings) I'd suggest to use your new Button inside of FooterActionButton where bold and primary is always set (composition always preferable in React).

Those areas are now untouched. But does it make sense to make your FooterActionButton use my Button?

> Offering className and style is a backdoor for hacky CSS patterns, I'd try to not allow these props except of when we can't get there yet as changing that now would lead to a large refactoring.
> ...

Will fix it. This PR is currently a draft.

> If possible we should try to use CSS modules from now on for every newly introduced or refactored styling. 
> We can do this here after rebasing this branch to master

This is already using `_delta-buttons.scss` if that's what you mean.

> I'd be up for not introducing any sub-folders like ui for components and rather keep them all on the same level. With a project of this size one searches anyhow more than looking through file-trees, and if one does so than I'm sure sometimes you'd expect things in ui which are not there or the other way around.

I didn't get the part from "With a project of this size...". Can you rephrase it please?

> Nice refactoring...

Told you, your way continues :)))

### CODE REVIEW

**Comment by adzialocha** [comment]
👋🏻 

> > For buttons inside DialogFooter (former DeltaDialogFooter) I've introduced the FooterActionButton (https://github.com/deltachat/deltachat-desktop/blob/master/src/renderer/components/Dialog/FooterActionButton.tsx) component which looks very similar to this one here.
> > Currently all buttons in dialogues are always bold and primary, so no need to be highly configurable in this context.
> > When rebasing this PR to master (with my refactorings) I'd suggest to use your new Button inside of FooterActionButton where bold and primary is always set (composition always preferable in React).
> 
> Those areas are now untouched. But does it make sense to make your FooterActionButton use my Button?

Since you already replaced the same parts as I did with your `Button`, I assume they are fairly similar and it would make sense to streamline the code.

I still think that using `Button` with the primary & bold style _inside_ of `FooterActionButton` makes sense, just to avoid duplicate code everywhere for the same style & functionality. We want a certain Button style for footers in dialogues (I couldn't find any exceptions), it is commonly used across the application in many places, then it makes sense to give it its own home.

> > If possible we should try to use CSS modules from now on for every newly introduced or refactored styling.
> > We can do this here after rebasing this branch to master
> 
> This is already using `_delta-buttons.scss` if that's what you mean.

`_delta-buttons.scss` is part of a larger `manifest.scss` document and therefore defined globally which is the opposite of the CSS module pattern.

Check out the implementations for `FooterActionButton` as an example on how to work with CSS modules, they are locally applied and scoped to components. Here is a nice introductions to CSS modules: https://css-tricks.com/css-modules-part-1-need/

> > I'd be up for not introducing any sub-folders like ui for components and rather keep them all on the same level. With a project of this size one searches anyhow more than looking through file-trees, and if one does so than I'm sure sometimes you'd expect things in ui which are not there or the other way around.
> 
> I didn't get the part from "With a project of this size...". Can you rephrase it please?

With this I wanted to say that this projects consists of hundreds of files, which is a fairly large size for a project.

When introducing sub-folders I think we need to be clear about the semantics of them. When `Button` is part of `ui`, why is not `Dialog` part of `ui` then? Isn't this also part of the user interface? etc. - to avoid these confusions altogether I think we can also just put everything into `components`.

**Comment by farooqkz** [comment]
adz's suggestions were addressed. requesting for review from Simon and adz.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 - Prefer light theme for the help and webxdc loading pages
 - Helper method to easily use confirmation dialogs #3601
 - Refactor using new `useConfirmationDialog` hook #3602
+- Refactor Buttons
```
Comment (adzialocha) [inline CHANGELOG.md:20]: PR number missing

**Review by adzialocha — APPROVED** [review]

**Comment by Simon-Laux** [comment]
Bildschirmfoto 2024-01-15 um 21 34 21
Bildschirmfoto 2024-01-15 um 21 35 06

dialog buttons have some problems in dark theme, those "clear" buttons should keep their hover style:
Bildschirmfoto 2024-01-15 um 21 37 10

The problem of missing space/margin in "new chat"-dialog seems to be older already, it's also on master, so no worries there.

**Review by Simon-Laux — CHANGES_REQUESTED** [review]
See my last comment, while code looks good to me at first glance the pr introduces problems with dark mode.

**Review by Simon-Laux — CHANGES_REQUESTED** [review]
Bildschirmfoto 2024-02-03 um 19 36 43

hover style is still broken

File: src/renderer/components/Button/style.module.scss [hunk 2]
```diff
@@ -0,0 +1,71 @@
+.button {
+  background-color: transparent;
+  border-color: transparent;
+  border-radius: 2px;
+  border-style: solid;
+  color: var(--bp4ButtonText);
```
[the remark this answers is not in the document]
Comment (Simon-Laux) [reply src/renderer/components/Button/style.module.scss:6]: or even if they are used elsewhere we could clone the variable, or we just do this in https://github.com/deltachat/deltachat-desktop/issues/3018

**Review by Simon-Laux — APPROVED** [review]
hover works now

File: src/renderer/components/Button/style.module.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (adzialocha) [reply src/renderer/components/Button/style.module.scss:6]: Nice to do a cleanup, but definitely something for another PR :+1:
